### PR TITLE
lu: Application framework

### PR DESCRIPTION
**Pull request by adamhicks**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Go
+
+on:
+  push:
+    branches: [ "**" ]
+  pull_request:
+    branches: [ master ]
```
Comment (echarrod) [inline .github/workflows/go.yml:7]: Looks like we have main as default branch 👍 
```suggestion
    branches: [ main ]
```

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,36 @@
 module github.com/luno/lu
 
 go 1.22.3
```
Comment (echarrod) [inline go.mod:3]: Just thinking - if we set 1.22.3 here, it means any users of the package would need that version at a minimum too.

Could we set to 1.21 for now, so it's compatible with more versions? 

Although I guess no one is using it yet, and 1.23 might be out next month, so probably fine 👍

**Review by echarrod — APPROVED** [review]
Nice

File: go.mod (hunk 2 above)
Comment (adamhicks) [reply go.mod:3]: i didn't make a conscious choice here, this is what i got from `go mod init ...`
i don't see a reason not to use 1.21 here though.

Comment (adamhicks) [reply go.mod:3]: found a reason, some recently added code uses cmp.Or, sticking to 1.22.3

Comment (echarrod) [reply go.mod:3]: All good, since 1.23 is out soon, I don't think this would be an issue anyway 👍

**Pull request merged**